lstchain/mc/plot_utils.py

```diff
@@ -76,144 +108,245 @@ def format_axes_array(ax, arr_i, arr_j, plot):
 
+    plot_MAGIC_sensitivity(ax)
     format_axes_sensitivity(ax)
-    ax.legend(numpoints=1,prop={'size':9},ncol=2,loc='upper right')
+    ax.legend(numpoints = 1, prop = {'size':9}, ncol = 2, loc = 'upper right')
+
+    return fig_sens
 
 def plot_positions_survived_events(df_gammas,
                                    df_protons,
                                    gammaness_g, gammaness_p,
-                                   theta2_g, p_contained, sens, E, eb, g, t):
+                                   theta2_g, p_contained, sensitivity, energy, 
+                                   n_bins_energy, gammaness_bins, theta2_bins, 
+                                   save_figure = False, ax=None):
+    """
+    Plot positions of surviving events after cuts
+
+    Parameters
+    --------
+    df_gammas: `pandas.DataFrame` gammas dl2 parameters
+    df_protons: `pandas.DataFrame` protons dl2 parameters
+    gammaness_g: `numpy.ndarray`  gammaness array of gamma events
+    gammaness_p: `numpy.ndarray`  gammaness array of proton events
+    theta2_g: `numpy.ndarray`  theta2 array of gamma events
+    p_contained: `numpy.ndarray`  containment of proton events inside
+                  the ring established in camera coordinates
+    sensitivity: `numpy.ndarray`  array with sensitivity values in energy bins
+    energy: `numpy.ndarray`  energy edge bins (size n_bins_energy + 1)
+    n_bins_energy: `int`  number of bins in energy
+    gammaness_bins: `numpy.ndarray`  gammaness bins
+    theta2_bins: `numpy.ndarray`  theta2 bins
+
+    Returns
+    --------
+
+    """
 
     e_reco_g = 10**df_gammas.mc_energy
     e_reco_p = 10**df_protons.mc_energy
-    for i in range(0,eb):
-        print(E[i], E[i+1])
-        ind = np.unravel_index(np.nanargmin(sens[i], axis=None), sens[i].shape)
-        events_g = df_gammas[(e_reco_g < E[i+1]) & (e_reco_g > E[i]) \
-                      & (gammaness_g > g[ind[0]]) & (theta2_g < t[ind[1]])]
-
-        events_p = df_protons[(e_reco_p < E[i+1]) & (e_reco_p > E[i]) \
-                      & (gammaness_p > g[ind[0]]) & p_contained]
+    for i in range(0, n_bins_energy):
+        fig, ax = plt.subplots()
+        print("Energy range [GeV]: ", energy[i], energy[i+1])
+        ind = np.unravel_index(np.nanargmin(sensitivity[i], axis = None), sensitivity[i].shape)
+        events_g = df_gammas[(e_reco_g < energy[i+1]) & (e_reco_g > energy[i]) \
+                      & (gammaness_g > gammaness_bins[ind[0]]) & (theta2_g < theta2_bins[ind[1]])]
+
+        events_p = df_protons[(e_reco_p < energy[i+1]) & (e_reco_p > energy[i]) \
+                      & (gammaness_p > gammaness_bins[ind[0]]) & p_contained]
         events_p.intensity.hist()
-        plt.xlabel("Log(10) Intensity Protons")
-        plt.savefig("intensity_prot%d" % i)
-        plt.show()
-        df = pd.concat([events_g, events_p], ignore_index=True)
+        ax.xlabel("Log(10) Intensity Protons")
+        ax.savefig("intensity_prot%d" % i)
```

Comment:
`ax.savefig` does not exist.
`fig.savefig`

Reply:
Thanks!
- Regarding the unit tests, I agree, but this PR has already become obsolete because it is taking too long (my bad...). It would be really nice if it can be merged so we can move on to a more robust implementation.
- For the `ctaplot` installation, as discussed in #131, since this is not an official package, I would prefer not to include it as a default dependency. A quick way of performing these plots just for comparison can be used/implemented in addition to the writing of the output for all the results for these figures of merit, that can be an input for `ctaplot` if people want to use it later.

Reply:
done